### DISCUSSION TITLE
Z-Wave+ CC V2 Lifeline reports and Security encapsulation

### DISCUSSION THREAD

**Discussion by AlCalzone** [zwave-js/zwave-js#1697] [General] [answered]
Hey @hanskroner - I have another case study!

In https://github.com/zwave-js/node-zwave-js/issues/1678 the user reports that he has 4 identical devices of which 1 reports its state automatically and 3 don't. All of them support Z-Wave+ CC V2, so they should be required to report instantly. All of them have the same association config, all of them are in short range to the controller.

The only difference I could find is that the one that does report is **not** included securely, all others are communication with S0 encapsulation.
At a first glance I couldn't find any mention in the specs that this changes things - but I'm almost sure it is somewhere in the fine print. Question is: Is zwave-js doing something wrong or should I tell the user to re-include his devices non-securely?

**Comment by hanskroner** [comment, marked as answer]
> Hey @hanskroner - I have another case study!
> 
> In #1678 the user reports that he has 4 identical devices of which 1 reports its state automatically and 3 don't. All of them support Z-Wave+ CC V2, so they should be required to report instantly. All of them have the same association config, all of them are in short range to the controller.

Z-Wave Plus v2 does indeed require Multi-level Switches to send a `MULTILEVEL-SWITCH_REPORT` to the Lifeline when the multi-level value changes - but only once the device has reached the final state.

> The only difference I could find is that the one that does report is **not** included securely, all others are communication with S0 encapsulation.
> At a first glance I couldn't find any mention in the specs that this changes things - but I'm almost sure it is somewhere in the fine print. Question is: Is zwave-js doing something wrong or should I tell the user to re-include his devices non-securely?

I don't think being included securely or not matters - the device should still report out to the lifeline as required (@Silabs-Crilles would know if I'm wrong about this). My impression is the device is misbehaving when included using S0. It would be interesting to see how it behaves when included with S2.
